feat: add php 8.4 support

pweyck opened the pull request:
<!--
Thank you for contributing to Shopware! Please fill out this description template to help us to process your pull request.

Please make sure to fulfill our contribution guideline (https://developer.shopware.com/docs/resources/guidelines/code/contribution?category=shopware-platform-dev-en/contribution).

Do your changes need to be mentioned in the documentation?
Please create a second pull request at https://github.com/shopware/docs
-->

### 1. Why is this change necessary?

Support php8.4 and test it in the CI

### 2. What does this change do, exactly?

Allow php8.4 in the composer constraint and fix some php 8.4 issues

### 3. Describe each step to reproduce the issue or behaviour.


### 4. Please link to the relevant issues (if any).
<!-- Examples:
- closes #123  - closes the issue #123 when the PR is merged
- relates #123 - relates to the issue #123

In case of issue existing only on Jira, link to the Jira issue.
- Jira issue: https://shopware.atlassian.net/browse/NEXT-123
-->

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfill them.

pweyck committed:
chore(ci): test with php 8.4

pweyck committed:
tmp: test expression

pweyck committed:
Merge remote-tracking branch 'origin/trunk' into test-with-php-8.4

pweyck committed:
fix: use correct ArrayParameterType::STRING for hex uuids

pweyck committed:
fix: use valid local in test

pweyck committed:
fix: use ArrayParameterType correctly

pweyck committed:
fix: style

pweyck commented on phpunit.xml.dist:
```diff
@@ -142,7 +142,7 @@
         <!--To see the full stackTrace of a Deprecation set the value to a regex matching the deprecation warning-->
         <!--https://symfony.com/doc/current/components/phpunit_bridge.html#display-the-full-stack-trace-->
-        <env name="SYMFONY_DEPRECATIONS_HELPER" value="ignoreFile=./deprecation.ignore" />
+        <env name="SYMFONY_DEPRECATIONS_HELPER" value="max[direct]=0&amp;ignoreFile=./deprecation.ignore" />
```
how can I generate a new one? Execute ALL unit tests with `SYMFONY_DEPRECATIONS_HELPER='generateBaseline=true&baselineFile=./deprecation.ignore'`?

pweyck committed:
fix: do not allow failure for php8.4

pweyck committed:
Merge remote-tracking branch 'origin/trunk' into test-with-php-8.4

pweyck committed:
chore: disable danger rule DisallowRepeatedCommits, we always use squash merge

pweyck commented:
> Fails
> 🚫 	You should not use the same commit message multiple times
> 	Warnings
> ⚠️ 	The Pull Request doesn't contain any changelog file

We don't need the duplicate commit message check because we use squash merge now 

https://github.com/shopware/shopware/pull/5977/files#diff-29d32130150acc4d0797220d4ddcebe5e076fd9212ecd3c118c1d19a31394643L39 should fix it in the future

mitelg reviewed (commented):
do not forget the composer.json files of the packages :wink:

pweyck committed:
test: test ats with php 8.2 and 8.4

pweyck committed:
test: test update instead of ats with php 8.4

pweyck commented:
> do not forget the composer.json files of the packages 😉

~I just thought we should add a test that uses the sub packages directly, but we actually already do that in the ATS update job. So I've added a job to the matrix with php 8.4~

That wont work for obvious reasons 🙈 

I've added an issue to our board. I think a simple test with the production template would be enough.

pweyck committed:
chore: allow php8.4 in sub composer packages
